### PR TITLE
Promises part 1

### PR DESCRIPTION
**Pull request by alyn509**

### CODE REVIEW

File: elrond-wasm/src/types/interaction/contract_call.rs [hunk 1]
```diff
@@ -172,15 +198,21 @@ where
                 let zero = BigUint::zero();
                 let endpoint_name = ManagedBuffer::new_from_bytes(ESDT_TRANSFER_FUNC_NAME);
 
+                let success_callback = b"";
+                let error_callback = b"";
```
Comment (dorin-iancu) [inline elrond-wasm/src/types/interaction/contract_call.rs:202]: Why is this being reset? Same in other methods.

Comment (alyn509) [reply elrond-wasm/src/types/interaction/contract_call.rs:202]: good catch :+1: 
forgot so solve the esdt transfer calls

Comment (alyn509) [reply elrond-wasm/src/types/interaction/contract_call.rs:202]: this should remain only for `new_with_esdt_payment`

Comment (alyn509) [reply elrond-wasm/src/types/interaction/contract_call.rs:202]: just in case I will also make a mandos test with some transfers to make sure all works

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**